src/Data/YAML/Dumper.hs

```diff
@@ -0,0 +1,107 @@
+{-# LANGUAGE Safe              #-}
+{-# LANGUAGE OverloadedStrings #-}
+
+
+module Data.YAML.Dumper
+    ( dumpYAML
+    , dumpYAML'
+    , dumpEvents
+    ) where
+
+import           Data.YAML.Event.Internal as YE
+import           Data.YAML.Internal       as YI
+import           Data.YAML.Schema         as YS
+import           Data.YAML.Event.Writer   (writeEvents)
+
+import qualified Data.ByteString.Lazy     as BS.L
+import           Data.Map                 as Map
+import qualified Data.Text                as T
+
+
+type EvList = [Either String Event]
+type Node2EvList = [Node ()] -> EvList
+
+-- | Dump YAML Nodes using specified UTF encoding to a lazy 'BS.L.ByteString'
+--
+-- @since 0.2.0
+dumpYAML ::  Encoding -> [Node ()] -> BS.L.ByteString
+dumpYAML encoding nodes = writeEvents encoding $ case sequence (dumpEvents nodes) of
+    Left str -> error str
+    Right ev -> ev
+
+-- | Convenience wrapper over 'dumpYAML' expecting exactly one YAML node
+--
+-- @since 0.2.0
+dumpYAML' ::  Encoding -> Node () -> BS.L.ByteString
```

Comment:
This should be the dual of `decodeNode'`

(i.e. this shall take a schema resolver argument; not necessarily this PR; can be done in a future PR)